### DISCUSSION TITLE
Mod downloads from servers take a long time

### DISCUSSION THREAD

**Discussion by RagingFlames** [FakeFishGames/Barotrauma#15365] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

When I host a server and my friends join, they often need to download mods from me (Even if we both manually update our mods using the in game menu). The download system is VERY slow, and it is not limited by my internet or my friends.  It seems to be hard limited at half a megabyte per second by the game

![image](https://github.com/user-attachments/assets/37251ace-00f9-47c7-a0d7-5420db2f3c48)

I'm trying to send the mod updates to 2-5 people at a time, the person with the slowest internet get about 50mb/s and I have gigabit internet.

### Reproduction steps

1. Start a multiplayer server with mods
2. Have people connect to the server without the mods
3

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.7.7.0 (Winter Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by RagingFlames** [comment]
Is there any word on this? Does anyone else also have this problem?

**Comment by cl0ck-byte** [comment]
The download speed is capped by p2p/SDR (steam relays) to like 200KB/s per player. No, the game is not limiting your bandwith, it's just the max bandwith of the p2p network.

Switch to Dedicated Server setup if you can, or update your mods and disable `AllowModDownloads` in `serversettings.xml` of your game to prevent anyone from downloading mods off you, so they're forced to use Workshop instead (needs a restart)

Note that disabling `AllowModDownloads` will boot off anyone who has outdated mods and no one will be able to join you whenever your mods are outdated.
